galaxy_api/api/ui/viewsets/namespace.py

```diff
@@ -75,6 +75,32 @@ def create(self, request, *args, **kwargs):
             return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+    def update(self, request, *args, **kwargs):
+        groups = []
+        for account in request.data['groups']:
+            if account == RH_PE_ACCOUNT_SCOPE:
+                groups.append(account)
```

Comment:
For a namespace "owned" by user "foo", this allows them to add  RH_PE_ACCOUNT_SCOPE  ('system:partner-engineers') to the list of groups. Is this to support replacing the groups list with the existing members (ie, partner-engineers + some new group)?

Can a user remove the PE group from the groups list?  It feels like the intention would be to not allow that, but I'm not sure what the code does.
If namespaces are created by default with a RH_PE_ACCOUNT_SCOPE group in the groups list, could PE groups just get removed out of the groups list?

Reply:
Would it make sense to require a permissions.IsPartnerEngineer permission if there is a PE group in the groups list?

Reply:
> For a namespace "owned" by user "foo", this allows them to add RH_PE_ACCOUNT_SCOPE ('system:partner-engineers') to the list of groups.

`RH_PE_ACCOUNT_SCOPE` is expected to be assigned to the namespace by default as requested in the [issue](https://github.com/ansible/galaxy-dev/issues/176#issue-521151758). The `system:partner-engineers` is automatically added to the groups listed by the user in UI ([here](https://github.com/ansible/ansible-hub-ui/pull/87/files#diff-f008b32a6b80ef0cb132f83160fb12d8R41)).

> Can a user remove the PE group from the groups list?

It is possible to edit the namespace and remove the default `PE` group.

>  It feels like the intention would be to not allow that, but I'm not sure what the code does.

I'm not sure about that. I've assumed that we want to have the PE group by default, but allow the partners to remove it.

@chouseknecht, is my assumption correct?

Reply:
> Would it make sense to require a permissions.IsPartnerEngineer permission if there is a PE group in the groups list?

@alikins Now, that we ensure the `system:partner-engineers` stays as a namespace owner (cabc3c6e3fdbf7b7887199b43717bc96fa611b90), is this still a concern?

Issue https://github.com/ansible/galaxy-dev/issues/176 states, that  "_Partner engineers and admins must be able to create and manage namespaces, including adding and removing owners"_.

Wouldn't this prevent admins to manage namespaces?
What are the scenarios we should try to prevent?

Reply:
@romanblanco hmm, I'm not sure. If it works, that works for me.

Reply:
/cc @cutwater @newswangerd